packages/toolpad-app/src/runtime/ToolpadApp.tsx

```diff
@@ -1106,18 +1106,27 @@ function RenderedPages({ pages, defaultPage }: RenderedPagesProps) {
   return (
     <Routes>
       {pages.map((page) => (
-        <Route
-          key={page.id}
-          path={`/pages/${page.id}`}
-          element={
-            <RenderedPage
-              nodeId={page.id}
-              // Make sure the page itself mounts when the route changes. This make sure all pageBindings are reinitialized
-              // during first render. Fixes https://github.com/mui/mui-toolpad/issues/1050
-              key={page.id}
-            />
-          }
-        />
+        <React.Fragment key={page.id}>
+          <Route
+            path={`/pages/${page.id}`}
+            element={
+              <RenderedPage
+                nodeId={page.id}
+                // Make sure the page itself mounts when the route changes. This make sure all pageBindings are reinitialized
+                // during first render. Fixes https://github.com/mui/mui-toolpad/issues/1050
+                key={page.id}
+              />
+            }
+          />
+        </React.Fragment>
+      ))}
+      {pages.map((page) => (
```

Comment:
We can also have this in the same mapping function as the pages for ids above if it's more efficient (I guess it's negligible though).

Reply:
The idea was that id always has precedence in case there's a naming conflict between a name and an id of two different pages

Reply:
I guess it would still work if the route for the name comes after the respective route for the id?
Anyway it's not a big deal, just pointing out the possibility.

Reply:
Ah, I misunderstood it, good thinking! Thanks for clearing it up.